paramiko/ssh_exception.py

```diff
@@ -114,18 +114,17 @@ class BadHostKeyException(SSHException):
     """
 
     def __init__(self, hostname, got_key, expected_key):
-        message = (
-            "Host key for server {} does not match: got {}, expected {}"
-        )  # noqa
-        message = message.format(
-            hostname, got_key.get_base64(), expected_key.get_base64()
-        )
-        SSHException.__init__(self, message)
+        SSHException.__init__(self, hostname, got_key, expected_key)
         self.hostname = hostname
         self.key = got_key
         self.expected_key = expected_key
-        # for unpickling
-        self.args = (hostname, got_key, expected_key)
+
+    def __str__(self):
+        return "Host key for server %s does not match: got %s, expected %s" % (
+            self.hostname,
+            self.got_key.get_base64(),
```

Comment:
just spotted a bug here (confusing arg vs prop name), hang on a sec

Reply:
squashed in the fix

Reply:
Argh I'm literally working on this myself right now, believe I caught this tho.